### DISCUSSION TITLE
NAPALM configuration

### DISCUSSION THREAD

**Discussion by Tola85** [netbox-community/netbox#11727] [Q&A]
Hello All, new to this and im trying to get napalm configuration working,

i have configured devices, ips and platform, but when i go to napalm tabs to pull configurations etc i get failures

i use cisco SG250X series and Cisco SX550X series.

have set up napalm username and password in configuration.py
napalm is installed

but get this error:
Error connecting to the device at 192.168.250.15: Authentication to device failed.
Common causes of this problem are:
1. Invalid username and password
2. Incorrect SSH-key file
3. Connecting to the wrong device
Device settings: cisco_ios 192.168.250.15:22
Bad authentication type; allowed types: ['']

i have tried to put additional arguments in config but did not worked
NAPALM_USERNAME = 'user'
NAPALM_PASSWORD = 'passs'
NAPALM_TIMEOUT = 60
NAPALM_ARGS = {
    'secret': NAPALM_PASSWORD,
	'login as': NAPALM_USERNAME,
	'port': 22,
	'allow_agent' :true 
    # Include any additional args here
}

also is there any examples of configuration of platforms to set username and password specific for every platform?

thx

**Comment by tolasws** [comment]
> Hello All, new to this and im trying to get napalm configuration working,
> 
> i have configured devices, ips and platform, but when i go to napalm tabs to pull configurations etc i get failures
> 
> i use cisco SG250X series and Cisco SX550X series.
> 
> have set up napalm username and password in configuration.py napalm is installed
> 
> but get this error: Error connecting to the device at 192.168.250.15: Authentication to device failed. Common causes of this problem are:
> 
> 1. Invalid username and password
> 2. Incorrect SSH-key file
> 3. Connecting to the wrong device
>    Device settings: cisco_ios 192.168.250.15:22
>    Bad authentication type; allowed types: ['']
> 
> i have tried to put additional arguments in config but did not worked NAPALM_USERNAME = 'user' NAPALM_PASSWORD = 'passs' NAPALM_TIMEOUT = 60 NAPALM_ARGS = { 'secret': NAPALM_PASSWORD, 'login as': NAPALM_USERNAME, 'port': 22, 'allow_agent' :true # Include any additional args here }
> 
> also is there any examples of configuration of platforms to set username and password specific for every platform?
> 
> thx

Oh i have done it with driver napalm_s350 but this only works for config from SX550X switches but not working for LLDP or status, and for SG250X does not work at all
